### DISCUSSION TITLE
[Feature proposal]: for else

### DISCUSSION THREAD

**Discussion by IPdotSetAF** [dotnet/csharplang#7830] [Language Ideas] [answered]
this could be an small feature for csharpLang but nice to have. the idea comes from python, it basically runs the `else` statement if the loops iteration is broken (`break;`).

the usage in C# would be like this:

```
for(statement ; boolean; statement)
   statement;
else
   statement;
```

or

```
for(statement ; boolean; statement)
{
   statement;
}
else
{
   statement;
}
```

A lambda version of this feature is available at [github.com/IPdotSetAF/ForElseCSharp](https://github.com/IPdotSetAF/ForElseCSharp/blob/master/ForElse/CSharpExtensions.cs)

**Comment by ufcpp** [comment, marked as answer]
https://github.com/dotnet/csharplang/discussions/4387
